### PR TITLE
Add option to test API with mock service

### PR DESCRIPTION
**Pull request by luelher**
- Add option to test with new mock service
- Mock service is un heroku in [Here](https://swagger-mocks.herokuapp.com/). Code is [here](https://github.com/e-gob/interoperabilidad-mocks)

### CODE REVIEW

**Comment by luelher** [comment]
Hi @leosoto @afiebig I want that you take a look too to the mock service project that was created to mock all response of the platform, and is used in this change.

https://github.com/e-gob/interoperabilidad-mocks

File: app/assets/javascripts/service_versions.coffee [hunk 1]
```diff
@@ -64,16 +64,7 @@ paramsFromEditors = ->
...
     params["#{loc}_params"] = JSON.parse(editor.getValue())
-  return params
-
-resizeEditors = ->
-  for location, editor of editors
-    editor.resize()
-
-paramsFromEditors = ->
-  params = {}
-  for loc, editor of editors
```
Comment (leosoto) [inline app/assets/javascripts/service_versions.coffee]: That removed code was just useless? Or did you implement that functionality in a way I can't see here?

File: app/models/service_version.rb [hunk 2]
```diff
@@ -292,6 +292,14 @@ def base_url
     schemes.first + '://' + host + base_path
   end
 
+  def base_url_mock
+    ENV['URL_MOCK_SERVICE'] + base_path_mock
+  end
+
+  def base_path_mock
+    URI.escape("#{self.organization.name}/#{self.service.name}/#{self.version_number}#{self.spec_with_resolved_refs['definition']['basePath'] || ''}")
```
Comment (leosoto) [inline app/models/service_version.rb]: I think this can be better expressed as:
```ruby
def base_path_mock
  self.url + self.base_path
end
```

Comment (leosoto) [reply app/models/service_version.rb]: (maybe with a slash as separator, dunno if the `url` or the `base_path` already has it)

File: app/models/service_version.rb [hunk 3]
```diff
@@ -304,22 +312,21 @@ def base_path
     self.spec_with_resolved_refs['definition']['basePath'] || ''
   end
 
-  def invoke(verb, path, path_params, query_params, header_params, raw_body)
+  def invoke(verb:, path:, path_params:, query_params:, header_params:, raw_body:, destination:)
```
Comment (leosoto) [inline app/models/service_version.rb]: Nice one! I see this was introduced on Ruby 2.1. I didn't know about it :)

File: app/models/service_version.rb [hunk 4]
```diff
@@ -304,22 +312,21 @@ def base_path
...
-
-      RestClient::Request.execute(
-        method: verb,
-        url: base_url + _resolve_path(path, path_params),
-        # TODO: Create RestClient::ParamsArray for arrays in query_params or they will be mangled with the [] suffix
-        #       and also pre-process arrays in headers, somehow (they aren't handled by restclient)
-        headers: header_params.merge(params: query_params),
-        payload: raw_body
-      )
+        RestClient::Request.execute(
```
Comment (leosoto) [inline app/models/service_version.rb]: Why this has extra indentation compared with the old code?

File: test/features/support/ui_test_helper.rb [hunk 5]
```diff
@@ -39,4 +39,10 @@ def expand_console_form(page)
     page.execute_script('$("#consoleForm a[data-toggle=\"collapse-next\"]").click()')
   end 
 
+  def select_test_with_mock_service
+    find(:css, ".console .btn.btn-primary.dropdown-toggle").click
+
+    find(:css, "#btns-service-console > li:nth-child(2)").click
```
Comment (leosoto) [inline test/features/support/ui_test_helper.rb]: Thanks so much for wrapping this into a readable method 👏

File: app/assets/javascripts/service_versions.coffee (hunk 1 above)
Comment (luelher) [reply app/assets/javascripts/service_versions.coffee]: I was duplicated in the same file

File: app/models/service_version.rb (hunk 2 above)
Comment (luelher) [reply app/models/service_version.rb]: Ok I'm going to change it

File: app/assets/javascripts/service_versions.coffee (hunk 1 above)
Comment (leosoto) [reply app/assets/javascripts/service_versions.coffee]: 👍

File: app/models/service_version.rb [hunk 6]
```diff
@@ -304,17 +316,24 @@ def base_path
     self.spec_with_resolved_refs['definition']['basePath'] || ''
   end
 
-  def invoke(verb, path, path_params, query_params, header_params, raw_body)
+  def invoke(options = {})
+    verb = options.fetch(:verb)
```
Comment (leosoto) [inline app/models/service_version.rb]: Why did you change this? The previous version with the Ruby 2.1 syntax looked way better!

File: app/models/service_version.rb [hunk 7]
```diff
@@ -292,6 +292,18 @@ def base_url
...
+  def url_mock
+    ENV['URL_MOCK_SERVICE']
+  end
+
+  def base_url_mock
+    url_mock + base_path_mock + base_path
+  end
+
+  def base_path_mock
+    URI.escape(self.organization.name) + '/' + self.service.name + '/' + self.version_number.to_s + '/'
```
Comment (leosoto) [inline app/models/service_version.rb]: I keep thinking that you don't need this. `self.url` should give you the same thing.

File: test/features/test_with_mock_option_test.rb [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    click_button "Crear Servicio"
+    assert_content page, "Servicio creado correctamente"
+
+    click_button "Probar Servicio"
+
+    select_test_with_mock_service
+
+    assert_content 'Servicio Simulado'
+
+    # TODO: Test Mock from here
```
Comment (leosoto) [inline test/features/test_with_mock_option_test.rb]: Will this stay as a TODO or do you plan to test the mock response?
